File: solutions/alfi.compitest/source_gen/alfi/compitest/Sequences/SeqRow.cpp
#include "SeqRow.h"
#include <alf/library/BasicInputOutput.hpp>
#include <string>
#include <memory>

namespace Sequences
{
  SeqRow::SeqRow(bool marked)
  {
    this->marked = marked;
  }
  void SeqRow::RowOperation()
  {
    alf::library::BasicInputOutput::WriteLine(std::string(">> RowOperation"));
  }
  bool SeqRow::IsMarkedEqual(const std::shared_ptr<SeqRow>& other)
  {
    return this->marked == other->marked;
  }
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/Sequences/SeqRow.cs
/// <filename>
///     SeqRow.cs
/// </filename>
using Alf.Library;
using System.Collections.Generic;

namespace Sequences
{
    public class SeqRow
    {

        public SeqRow(bool marked)
        {
            this.marked = marked;
        }

        public System.Collections.Generic.List<string> values = new System.Collections.Generic.List<string>();

        public bool marked;

        public virtual void RowOperation()
        {
            BasicInputOutput.WriteLine(">> RowOperation");
        }

        public virtual bool IsMarkedEqual(Sequences.SeqRow other)
        {
            return this.marked == other.marked;
        }
    }
}
